### PR TITLE
Increase memory in fastp and GroupByUMIs tasks

### PR DESCRIPTION
**Pull request by takutosato**
To address the out of memory issues we've observed. This solution should fix the errors but is far from ideal. The better approach is to change the memory variable at run time, either by:

a) Out of memory retry. This requires that we be able to check the "retry with more memory" box in terra (as detailed here, bullet point number 2: https://support.terra.bio/hc/en-us/articles/4403215299355). We can do this in the development terra workspace (https://app.terra.bio/#workspaces/broad-dsp-spec-ops-fc/Twist%20TCAP%20Data%20Processing%20DEV), but I'm not sure if we can do this in production.

b) Get a sense of how much memory each task needs in terms of the size of the bam, come up with a formula, then request this much memory at each run. I'm collecting some data on this.

### CODE REVIEW

File: tasks/broad/RNAWithUMIsTasks.wdl [hunk 1]
```diff
@@ -98,7 +98,7 @@ task Fastp {
     File adapter_fasta = "gs://gcp-public-data--broad-references/RNA/resources/Illumina_adapters.fasta"
 
     String docker = "us.gcr.io/broad-gotc-prod/fastp:1.0.0-0.20.1-1649253500"
-    Int memory_mb =  "16384"
+    Int memory_mb =  "32000"
```
Comment (wjdingman1) [inline tasks/broad/RNAWithUMIsTasks.wdl:101]: To fix your issue you need to determine this programatically based on the size of the input files + some buffer

We regularly have tasks in skylab and whole genome pipelines that use 100s of gb of memory so its totally fine to scale up

File: tasks/broad/RNAWithUMIsTasks.wdl [hunk 2]
```diff
@@ -600,7 +600,7 @@ task GroupByUMIs {
 
     String docker = "us.gcr.io/broad-gotc-prod/umi_tools:1.0.0-1.1.1-1638821470"
     Int cpu = 2
-    Int memory_mb = 7500
+    Int memory_mb = 1600
```
Comment (wjdingman1) [inline tasks/broad/RNAWithUMIsTasks.wdl:603]: did you mean to decrease memory here?

**Comment by wjdingman1** [comment]
add to whitelist

File: tasks/broad/RNAWithUMIsTasks.wdl (hunk 2 above)
Comment (takutosato) [reply tasks/broad/RNAWithUMIsTasks.wdl]: Yup, fixed

File: tasks/broad/RNAWithUMIsTasks.wdl (hunk 1 above)
Comment (takutosato) [reply tasks/broad/RNAWithUMIsTasks.wdl:101]: Yea I'm running some tests to figure out what this formula should be in our case.

**Comment by wjdingman1** [comment]
@takutosato the smart tests are failing when running the `CompareTranscriptomeBam` task, is this still the nondeterminism issue?

see here -> gs://broad-gotc-dev-cromwell-execution/VerifyRNAWithUMIs/725496b2-8cc2-49fb-92d8-d5c3c2f291d9/call-CompareTranscriptomeBam/stdout

**Comment by wjdingman1** [comment]
@takutosato @mmorgantaylor 

What is the status of the PR? I have comments above about a failing test, would like to get this  merged in if its ready but need to know where its at, thanks!

**Comment by takutosato** [comment]
@nikellepetrillo I hear Wes is no longer at the Broad. Could you help me with this? I'm trying to figure out why the tests are failing.

File: pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+# 1.0.17
+2022-07-21 (Date of Last Commit)
```
Comment (takutosato) [inline pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md:2]: @mmorgantaylor, can I just append my changes to this log of yours? (1.0.17)?

Comment (takutosato) [reply pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md:2]: And update the date?

Comment (takutosato) [reply pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md:2]: Actually I should leave the date as is

File: pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
+# 1.0.17
+2022-07-28 (Date of Last Commit)
+
+<<<<<<< HEAD
```
Comment (gbggrant) [inline pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md:4]: Artifact of a merge? Please remove.

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.changelog.md [hunk 5]
```diff
@@ -1,3 +1,11 @@
+# 1.0.8
+* Specify the RSEM post-processed transcriptome bam as output
```
Comment (gbggrant) [inline pipelines/broad/rna_seq/RNAWithUMIsPipeline.changelog.md:2]: Please add the date / and date of last commit line here (above the list of changes).

**Review by gbggrant — COMMENTED** [review]
two minor things need to be fixed in changelogs. Otherwise looks good.

File: pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md (hunk 4 above)
Comment (takutosato) [reply pipelines/broad/internal/rna_seq/BroadInternalRNAWithUMIs.changelog.md:4]: Yes, done

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.changelog.md (hunk 5 above)
Comment (takutosato) [reply pipelines/broad/rna_seq/RNAWithUMIsPipeline.changelog.md:2]: Done

**Review by gbggrant — APPROVED** [review]

File: tasks/broad/RNAWithUMIsTasks.wdl [hunk 6]
```diff
@@ -98,11 +98,14 @@ task Fastp {
     File adapter_fasta = "gs://gcp-public-data--broad-references/RNA/resources/Illumina_adapters.fasta"
 
     String docker = "us.gcr.io/broad-gotc-prod/fastp:1.0.0-0.20.1-1649253500"
-    Int memory_mb =  "16384"
+    Int memory_mb =  ceil(1.5*size(fastq1, "MiB")) + 8192 # Experimentally determined formula for memory allocation
     Int disk_size_gb = 5*ceil(size(fastq1, "GiB")) + 128
+    File monitoring_script = "gs://broad-dsde-methods-monitoring/cromwell_monitoring_script.sh"
```
Comment (jessicaway) [inline tasks/broad/RNAWithUMIsTasks.wdl:103]: I know we need to add the monitoring script this way in order to use it on Terra, but is this script at least publicly accessible? If not can we use a public copy? (same comment applies to other instances as well)

File: tasks/broad/RNAWithUMIsTasks.wdl [hunk 7]
```diff
@@ -600,24 +606,31 @@ task GroupByUMIs {
 
     String docker = "us.gcr.io/broad-gotc-prod/umi_tools:1.0.0-1.1.1-1638821470"
     Int cpu = 2
-    Int memory_mb = 7500
+    # Int memory_mb = ceil(1.2 * size([bam, bam_index], "GiB")) + 4096 # Using the empirically estimated slope of (max memory usage) as a function of (input disk size)
```
Comment (jessicaway) [inline tasks/broad/RNAWithUMIsTasks.wdl:609]: Is this code still needed? If not please clean up

**Review by jessicaway — APPROVED** [review]
Looks good, just a couple minor comments

File: tasks/broad/RNAWithUMIsTasks.wdl (hunk 6 above)
Comment (takutosato) [reply tasks/broad/RNAWithUMIsTasks.wdl:103]: I don't think it's publicly accessible as is; is there a place where we can put it?

File: tasks/broad/RNAWithUMIsTasks.wdl (hunk 7 above)
Comment (takutosato) [reply tasks/broad/RNAWithUMIsTasks.wdl:609]: cleaned up

File: tasks/broad/RNAWithUMIsTasks.wdl (hunk 6 above)
Comment (jessicaway) [reply tasks/broad/RNAWithUMIsTasks.wdl:103]: Good question :) I'm not sure. I'll have to figure that out. This should be OK in the meantime

**Pull request merged**